tt_metal/impl/device/multi_device.cpp

```diff
@@ -53,17 +62,29 @@ std::vector<Device*> DeviceMesh::get_devices() const
 const DeviceIds DeviceMesh::get_device_ids() const
 {
     DeviceIds device_ids;
-    for (const auto& [device_id, device] : managed_devices) {
+    for (const auto& [device_id, device] : mesh_devices) {
         device_ids.push_back(device_id);
     }
     return device_ids;
 }
 
 int DeviceMesh::num_devices() const
 {
-    return managed_devices.size();
+    return mesh_devices.size();
 }
 
+void DeviceMesh::close_devices() {
+    // TODO: change api to a yield, shouldn't allow closing sub grids in a pool of devices
+    tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);
```

Comment:
Are there any issues if those `tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(false);` is called twice (i.e. explicit close_devices() and destructor)? Or should we guard this by checking if `not managed_devices.empty()`